### PR TITLE
Update MFEngine models to contain attributes from pydantic models

### PR DESCRIPTION
**Pull request by courtneyholcomb**
resolves #SEM-202

### Description
Updates `MFEngine` `Metric` and `Dimension` to include the attributes available on their corresponding pydantic models.
This changes the behavior of `simple_dimensions_for_metrics` slightly, since the primary time dim will show up once for each relevant semantic model (instead of just once, total).
This is needed in order to expose metadata queries in MF server with more robust information - specifically `list_metrics` and `list_dimensions`.

<!--- 
  Before requesting review, please make sure you have:
  1. read [the contributing guide](https://github.com/dbt-labs/metricflow/blob/main/CONTRIBUTING.md),
  2. signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
  3. run `changie new` to [create a changelog entry](https://github.com/dbt-labs/metricflow/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: metricflow/engine/models.py [hunk 1]
```diff
@@ -1,19 +1,35 @@
 from __future__ import annotations
 
 from dataclasses import dataclass
-from typing import List
+from typing import List, Optional
+
+from dbt_semantic_interfaces.objects.metric import MetricType, MetricTypeParams, Metric as PydanticMetric
```
Comment (tlento) [inline metricflow/engine/models.py:6]: Might be worth calling PydanticMetric something like SemanticManifestMetric or similar. It's so long though. ModelMetric would've been nice. Sigh.

**Review by tlento — APPROVED** [review]
Seems reasonable!

**Comment by tlento** [comment]
Oh can you run `changie new` and add an Under the Hood entry? I don't know if Under the Hood is appropriate but I think it's fine.

File: metricflow/engine/models.py (hunk 1 above)
Comment (courtneyholcomb) [reply metricflow/engine/models.py:6]: I don't mind long!

**Pull request merged**